frameProcessor/src/KafkaProducerPlugin.cpp

```
@@ -149,6 +140,17 @@ namespace FrameProcessor {
       this->max_q_buffer_time_ = config.get_param<std::string>(CONFIG_KAFKA_MAX_Q_BUFFER_TIME);
     }
 
+    bool need_configure_topic = false;
+    if (config.has_param(CONFIG_SERVERS)) {
+      destroy_kafka();
+      configure_kafka_servers(config.get_param<std::string>(CONFIG_SERVERS));
+      need_configure_topic = true;
+    }
+
+    if (config.has_param(CONFIG_TOPIC) || need_configure_topic) {
+      configure_kafka_topic(config.get_param<std::string>(CONFIG_TOPIC));
+    }
+
```

Why does this need to come after the parameters? Do we need to force a reconfigure of the server if certain other things change?

---

The producer needs to be created before the topic can be configured, so configure_kafka_servers has to come first to set the parameters and create the producer, then the topic can be configured. 

Any time kafka configuration parameters change you need to first create producer then configure the topic, but there should be no reason to change parameters after start up. 

@hir12111 might want to expand on that.

---

We are assuming that everything is configured at start, the parameters are used when creating the kafka handler, that's why we need to first set the parameters (as attributes), and then configure the server so that those parameters are used.

---

If we are able to change them at runtime, they should take effect without having to configure the server explicitly. My initial thought is that it should raise an error if you try to configure them after startup, but Emilio suggests that it could be useful during commissioning to be able to tune these parameters at runtime, so maybe we should allow it. I think there should be a need_configure_server flag that is set by when the relevant parameters are changed, as with the topic.